configs/example/gem5_library/riscv-demo-board-se.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2021 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+This script uses the RiscvDemoBoard to run a simulation.
+
+Usage:
+
+(in gem5 directory)
+
+scons build/ALL/gem5.fast
+build/ALL/gem5.fast configs/example/gem5_library/riscv-demo-board-se.py --workload=riscv-matrix-multiply-run --version=1.0.0
+
+"""
+
+import argparse
+
+import m5
+
+from gem5.prebuilt.demo.riscv_demo_board import RiscvDemoBoard
+from gem5.resources.resource import obtain_resource
+from gem5.simulate.exit_event import ExitEvent
+from gem5.simulate.simulator import Simulator
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("--workload")
+
+parser.add_argument("--version")
+
+args = parser.parse_args()
+
+# Here we set up the board. The prebuilt RiscvDemoBoard allows for FS
+# or SE simulation.
+board = RiscvDemoBoard(is_fs=False)
+
+# Here we set a workload. If nothing is passed for resource_version, the
+# latest resource version compatible with the gem5 version will be obtained.
+board.set_workload(
+    obtain_resource(resource_id=args.workload, resource_version=args.version)
+)
```

Comment:
why do you need to declare `is_fs` in the `RiscvDemoBoard` constructor? Can't this just be infered by what workload is passed? I.e., if a FS is one is called then setup teh FS board, if not, setup the SE workload.

If you did that you could just merge these three demos board script together into one